x-pack/plugins/siem/public/components/page/network/domains_table/first_last_seen/index.test.tsx

```diff
@@ -0,0 +1,211 @@
+    );
+  });
+
+  test('First Seen', async () => {
+    const { container } = render(
+      <TestProviders>
+        <MockedProvider mocks={mockFirstLastSeenDomainQuery} addTypename={false}>
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={FlowTarget.source}
+            type="first-seen"
+          />
+        </MockedProvider>
+      </TestProviders>
+    );
+
+    await wait();
+
+    expect(container.innerHTML).toBe(
+      '<span class="euiToolTipAnchor">Apr 8, 2019 @ 16:09:40.692</span>'
+    );
+  });
+
+  test('Last Seen', async () => {
+    const { container } = render(
+      <TestProviders>
+        <MockedProvider mocks={mockFirstLastSeenDomainQuery} addTypename={false}>
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={FlowTarget.source}
+            type="last-seen"
+          />
+        </MockedProvider>
+      </TestProviders>
+    );
+    await wait();
+    expect(container.innerHTML).toBe(
+      '<span class="euiToolTipAnchor">Apr 8, 2019 @ 18:35:45.064</span>'
+    );
+  });
+
+  test('First Seen is empty but not Last Seen', async () => {
+    const badDateTime = cloneDeep(mockFirstLastSeenDomainQuery);
+    // tslint:disable-next-line:no-any
+    (badDateTime[0] as any).result.data.source.DomainFirstLastSeen.firstSeen = null;
+    const { container } = render(
+      <TestProviders>
+        <MockedProvider mocks={badDateTime} addTypename={false}>
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={FlowTarget.source}
+            type="last-seen"
+          />
+        </MockedProvider>
+      </TestProviders>
+    );
+
+    await wait();
+
+    expect(container.innerHTML).toBe(
+      '<span class="euiToolTipAnchor">Apr 8, 2019 @ 18:35:45.064</span>'
+    );
+  });
+
+  test('Last Seen is empty but not First Seen', async () => {
+    const badDateTime = cloneDeep(mockFirstLastSeenDomainQuery);
+    // tslint:disable-next-line:no-any
+    (badDateTime[0] as any).result.data.source.DomainFirstLastSeen.lastSeen = null;
+    const { container } = render(
+      <TestProviders>
+        <MockedProvider mocks={badDateTime} addTypename={false}>
+          <FirstLastSeenDomain
+            ip={ip}
+            domainName={domainName}
+            flowTarget={FlowTarget.source}
+            type="first-seen"
+          />
+        </MockedProvider>
+      </TestProviders>
+    );
+
+    await wait();
+
+    expect(container.innerHTML).toBe(
+      '<span class="euiToolTipAnchor">Apr 8, 2019 @ 16:09:40.692</span>'
+    );
+  });
+
+  test('First Seen With a bad date time string', async () => {
+    const badDateTime = cloneDeep(mockFirstLastSeenDomainQuery);
+    // tslint:disable-next-line:no-any
+    (badDateTime[0] as any).result.data.source.DomainFirstLastSeen.firstSeen = 'something-invalid';
```

Comment:
Optional: This is a bit of tech debt now, because that mock isn't typed. You could type it and then remove this any if you want. Otherwise we are making more tests with `any`

Reply:
Cleaned it up -- now typed and `any`'s removed! 🧹

Added the following type for `MockFirstLastSeenDomainQuery`:

``` typescript
interface MockFirstLastSeenDomainQuery {
  request: {
    query: GetDomainFirstLastSeenQuery.Query;
    variables: GetDomainFirstLastSeenQuery.Variables;
  };
  result: {
    data?: {
      source: {
        id: string;
        DomainFirstLastSeen: {
          firstSeen: string | null;
          lastSeen: string | null;
        };
      };
    };
    errors?: [{ message: string }];
  };
}
```